apps/designer/app/designer/features/style-panel/controls/combobox-control.tsx

```diff
@@ -0,0 +1,145 @@
+import { KeyboardEvent } from "react";
+import type { StyleValue } from "@webstudio-is/react-sdk";
+import {
+  Flex,
+  Grid,
+  Text,
+  Combobox,
+  IconButton,
+  TextField,
+  Tooltip,
+} from "@webstudio-is/design-system";
+import { PropertyName } from "./property-name";
+import { getFinalValue } from "../shared/get-final-value";
+import { useIsFromCurrentBreakpoint } from "../shared/use-is-from-current-breakpoint";
+import { ControlProps } from "../style-sections";
+import { RowGap, ColumnGap } from "@webstudio-is/icons";
+
+const Unit = ({ value }: { value: StyleValue }) => {
+  if (value.type !== "unit" || value.unit === "number") return null;
+  return (
+    <Text
+      css={{
+        fontSize: "$1",
+        cursor: "default",
+      }}
+    >
+      {value.unit}
+    </Text>
+  );
+};
+
+const ComboboxControl = ({
+  currentStyle,
+  inheritedStyle,
+  setProperty,
+  styleConfig,
+  category,
+}: ControlProps) => {
+  const isCurrentBreakpoint = useIsFromCurrentBreakpoint(styleConfig.property);
+
+  const value = getFinalValue({
+    currentStyle,
+    inheritedStyle,
+    property: styleConfig.property,
+  });
+
+  if (value === undefined) return null;
+
+  const setValue = setProperty(styleConfig.property);
+
+  switch (styleConfig.property) {
```

Comment:
we need a code organization that scales to support all properties

Reply:
I guess I don't understand what is being specifically fixed here. for columnGap or rowGap, but it definitely looks like it belongs somewhere else